pkg/apis/openebs/v1alpha1/crds.go

```diff
@@ -0,0 +1,90 @@
+package v1alpha1
+
+import (
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// DiskResourceKind is the kind of Disk CRD
+	DiskResourceKind = "Disk"
+	// DiskResourceListKind is the list kind for Disk
+	DiskResourceListKind = "DiskList"
+	// DiskResourcePlural is the plural form used for disk
+	DiskResourcePlural = "disks"
+	// DiskResourceShort is the short name used for disk CRD
+	DiskResourceShort = "disk"
+	// DiskResourceName is the name of the disk resource
+	DiskResourceName = DiskResourcePlural + "." + GroupName
+
+	// BlockDeviceResourceKind is the kind of block device CRD
+	BlockDeviceResourceKind = "BlockDevice"
+	// BlockDeviceResourceListKind is the list kind for block device
+	BlockDeviceResourceListKind = "BlockDeviceList"
+	// BlockDeviceResourcePlural is the plural form used for block device
+	BlockDeviceResourcePlural = "blockdevices"
+	// BlockDeviceResourceShort is the short name used for block device CRD
+	BlockDeviceResourceShort = "bd"
+	// BlockDeviceResourceName is the name of the block device resource
+	BlockDeviceResourceName = BlockDeviceResourcePlural + "." + GroupName
+
+	// BlockDeviceClaimResourceKind is the kind of block device claim CRD
+	BlockDeviceClaimResourceKind = "BlockDeviceClaim"
+	// BlockDeviceClaimResourceListKind is the list kind for block device claim
+	BlockDeviceClaimResourceListKind = "BlockDeviceClaimList"
+	// BlockDeviceClaimResourcePlural is the plural form used for block device claim
+	BlockDeviceClaimResourcePlural = "blockdeviceclaims"
+	// BlockDeviceClaimResourceShort is the short name used for block device claim CRD
+	BlockDeviceClaimResourceShort = "bdc"
+	// BlockDeviceClaimResourceName is the name of the block device claim resource
+	BlockDeviceClaimResourceName = BlockDeviceClaimResourcePlural + "." + GroupName
+)
+
+func buildCRD(name, kind, listKind, plural, short string, scope apiext.ResourceScope) *apiext.CustomResourceDefinition {
```

Comment:
These CRDs are missing custom columns for display. Also in the future if we want to add a few more details like schema validation, this method via parameters can become a bottleneck. Either we can modify this to a builder patter or make use of the YAML string -> generic kubeclient to load the CRD into K8s, like we do in the maya/pkg/install.

Reply:
Thanks for changing this!